### ISSUE TITLE
Failures on sphgtod_t and sphgettime_t

### ISSUE THREAD

**Issue by rzinsly** [sphde/sphde#56]
We notice that the tests sphgtod_t and sphgettime_t may fail on some systems:

```
starting sphgettime value is      14344187166167083 nanoseconds
starting clock_gettime value is 14344187166164686 nanoseconds
sphgettime takes 22.333542 nanoseconds
Ending sphgettime value is       14344187389549943 nanoseconds
Ending clock_gettime value is    14344187389539295 nanoseconds
The difference between clock_gettime and sphgettime (10648ns)
exceeded the allowable drift of 9000 ns
```

These tests may not be correct as they assume that the OS won't switch tasks and the compiler will keep both calls close to each other.

**Comment by munroesj52** [comment]
So I assume this only is reporting for PPC and AT. If not I need to know the details.

yup. This test will fail it run on highly virtuallized (and heavily loaded) systems or QEMU/Mambo. 

I think this can only fail where the platform has a user-level timebase that is enable (ei ppc). Otherwise we are comparing the clock_gettime to clock_gettime :-/

So we can either make this a soft-fail or internally rerun the test (with sched_yield between) in hope to get a lucky quantum.

And if retry fails that implies a bad timebase conversion factor that I need to understand.

**Comment by rzinsly** [comment]
I got these errors on ppc64le servers without using AT, they are indeed virtuallized systems.

IMO making it a soft-fail is fine.

**Comment by munroesj52** [comment]
Ok I can change the failure return to 77 (SKIP). (the conditions for success are not meet).

It seems like XPASS (expect success so recheck) would be appropriate. But this requires more autoconf/automake changes, I want to avoid for now.

**Comment by munroesj52** [comment]
We can try returning 77 for sphgtod_t and sphgettime_t.

**Comment by rzinsly** [comment]
Fixed by #65.
